File: Largest_Number_At_Least_Twice_of_Others/dominantIndex.c
int dominantIndex(int* nums, int numsSize) {
    int maxOne = 0, maxTwo = 0, index = 0, i;
    
    for (i = 0; i < numsSize; i++) {
        if (maxOne < nums[i]) {
            maxTwo = maxOne;
            maxOne = nums[i];
            index = i;
        } else if (maxTwo < nums[i]) {
            maxTwo = nums[i];
        }
    }
    return maxOne >= maxTwo * 2 ? index : -1;
}

File: Largest_Number_At_Least_Twice_of_Others/dominantIndex.py
class Solution(object):
    def dominantIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0: return -1
        
        maxOne = maxTwo = 0
        index = -1
        
        for i, n in enumerate(nums):
            if maxOne < n:
                maxTwo = maxOne
                maxOne = n
                index = i
            elif maxTwo < n:
                maxTwo = n
                
        if maxOne < maxTwo*2:
            index = -1
            
        return index
